tests/model_serving/model_server/keda/conftest.py

```diff
@@ -0,0 +1,232 @@
+                        + "'\
+                                ,model_name='"
+                        + request.param["name"]
+                        + "'}",
+                        "namespace": model_namespace.name,
+                        "authModes": "bearer",
+                    },
+                    "authenticationRef": {
+                        # https://github.com/opendatahub-io/odh-model-controller/blob/6c83a07e764be8a6bb77b1b36589ed029605b186/internal/controller/serving/reconcilers/kserve_keda_reconciler.go#L47
+                        "name": "inference-prometheus-auth",
+                    },
+                    "target": {"type": "Value", "value": "2"},
+                },
+            }
+        ]
+    }
+
+    with create_isvc(**isvc_kwargs, teardown=False) as isvc:
+        yield isvc
+
+
+@pytest.fixture(scope="class")
+def ovms_keda_inference_service(
+    request: FixtureRequest,
+    unprivileged_client: DynamicClient,
+    unprivileged_model_namespace: Namespace,
+    ovms_kserve_serving_runtime: ServingRuntime,
+    ci_endpoint_s3_secret: Secret,
+) -> Generator[InferenceService, Any, Any]:
+    with create_isvc(
+        client=unprivileged_client,
+        name=f"{request.param['name']}-raw",
+        namespace=unprivileged_model_namespace.name,
+        external_route=True,
+        runtime=ovms_kserve_serving_runtime.name,
+        storage_path=request.param["model-dir"],
+        storage_key=ci_endpoint_s3_secret.name,
+        model_format=ModelAndFormat.OPENVINO_IR,
+        deployment_mode=KServeDeploymentType.RAW_DEPLOYMENT,
+        model_version=request.param["model-version"],
+        min_replicas=1,
+        max_replicas=5,
+        autoscaler_mode="keda",
+        auto_scaling={
+            "metrics": [
+                {
+                    "type": "External",
+                    "external": {
+                        "metric": {
+                            "backend": "prometheus",
+                            "serverAddress": THANOS_QUERIER_ADDRESS,
+                            "query": "ovms_requests_success{namespace='"
+                            + unprivileged_model_namespace.name
+                            + "'\
+                                ,model_name='"
+                            + f"{request.param['name']}-raw"
+                            + "'}",
+                            "namespace": unprivileged_model_namespace.name,
+                            "authModes": "bearer",
+                        },
+                        "authenticationRef": {
+                            # https://github.com/opendatahub-io/odh-model-controller/blob/6c83a07e764be8a6bb77b1b36589ed029605b186/internal/controller/serving/reconcilers/kserve_keda_reconciler.go#L47
+                            "name": "inference-prometheus-auth",
+                        },
+                        "target": {"type": "Value", "value": "50"},
+                    },
+                }
+            ]
+        },
```

Comment:
this is duplicate code in terms of creating the dict; create a function which returns the metrics dict